### ISSUE TITLE
Save training information

### ISSUE THREAD

**Issue by Tiamaty** [Tiamaty/ProjectManhattan#6]
Right now there is no way to check how the training went. Add a new data structure that saves the most important information in the training:

1. Cost function used to halt the training
2. The training algorithm used
3. How much time the training took
4. The values in each epoch of the cost function in the training set
5. The values in each epoch of the cost function in the validation set
6. The values in each epoch of the cost function in the test set
7. The number of epochs that the training took

**Comment by Tiamaty** [comment]
Done
